### PR TITLE
feat: add ThreeJS boilerplate with sample shader

### PR DESCRIPTION
**Pull request by thilan-tran**
## Demo

<div align="center">

</div>

## Changes

- Added new component `PlaylistVisual.tsx` with ThreeJS boilerplate and a sample shader
  - Takes height / width as parameters in order to use as hero visual
  - Demo at `http://localhost/shader-demo`

### CODE REVIEW

File: components/PlaylistVisual.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React, {useRef, useEffect} from 'react';
+import * as THREE from 'three';
+import {sampleShader} from '../utils';
+
+interface Props {
```
Comment (BryanPan342) [inline components/PlaylistVisual.tsx]: nit

```suggestion
interface PlaylistVisualProps {
```

File: components/PlaylistVisual.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    window.addEventListener('mousemove', handleMouseMove, false);
+
+    function animate() {
+      requestAnimationFrame(animate);
+      uniforms.u_time.value = clock.getElapsedTime();
+      renderer.render(scene, camera);
+    }
+    animate();
+
+    return () => ref.current.removeChild(renderer.domElement);
```
Comment (BryanPan342) [inline components/PlaylistVisual.tsx]: lets remove the event listener on unmount too

File: pages/shader-demo.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import PlaylistVisual from '../components/PlaylistVisual';
+
+function ShaderDemo(): JSX.Element {
+  return (
+    <Layout>
+      <div style={{
+        height: '100vh',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
```
Comment (BryanPan342) [inline pages/shader-demo.tsx:12]: he can center a div 😫

**Review by BryanPan342 — DISMISSED** [review]
LGTM!! I have a couple of nits that you can choose to resolve if you want!

If not, just remove the `pr/do-not-merge` tag from the PR and mergify will auto merge/squash ur changes!

Also, is it possible to get load time?

File: components/PlaylistVisual.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      uniforms.u_mouse.value.set(evt.pageX, w - evt.pageY);
+    }
+    window.addEventListener('mousemove', handleMouseMove, false);
+
+    function animate() {
+      requestAnimationFrame(animate);
+      uniforms.u_time.value = clock.getElapsedTime();
+      renderer.render(scene, camera);
+    }
+    animate();
```
Comment (BryanPan342) [inline components/PlaylistVisual.tsx:56]: wait where did the return function go HAHA

**Review by BryanPan342 — DISMISSED** [review]
wait wat?

i was thinking like 

```ts
return () => {
  ref.current.removeChild(renderer.domElement);
  window.removeEventListener('mousemove', handleMouseMove, false);
}
```

File: components/PlaylistVisual.tsx (hunk 4 above)
Comment (thilan-tran) [reply components/PlaylistVisual.tsx:56]: Ooh my bad, misunderstood what you were asking for 😅

**Review by BryanPan342 — APPROVED** [review]
KING SHIT 👑💩

**Pull request merged**